### PR TITLE
Define CNOT as a pass-thru to an implementation specified in the config file

### PR DESCRIPTION
**Pull request by blakejohnson**

### CODE REVIEW

File: QGL/PulsePrimitives.py [hunk 1]
```diff
@@ -681,6 +677,21 @@ def CNOT_CR(controlQ, targetQ, **kwargs):
...
 
+def CNOT_simple(source, target, **kwargs):
+    # construct (source, target) channel and pull parameters from there
+    channel = ChannelLibrary.EdgeFactory(source, target)
+    channel.pulseParams['piAmp'] = channel.pulseParams['amp']
+    p = X(channel, **kwargs)
+    return p._replace(label="CNOT")
+
+try:
+    cnot_impl = globals()[config.cnot_implementation]
```
Comment (caryan) [inline QGL/PulsePrimitives.py]: If you've imported config why do you need the `globals()`?

Comment (caryan) [inline QGL/PulsePrimitives.py]: Uh, sorry I wasn't following the code properly; it's fine.

File: QGL/config.py [hunk 2]
```diff
@@ -32,3 +32,7 @@
 
 # select pulse library (standard or all90)
 pulse_primitives_lib = QGLCfg.get('PulsePrimitivesLibrary', 'standard')
+
+# select a CNOT implementation (a name of a Pulse function that implements
+# CNOT in your gate set)
+cnot_implementation = QGLCfg.get('cnot_implementation', 'CNOT_CR')
```
Comment (caryan) [inline QGL/config.py]: Default to `CNOT_simple`?

File: QGL/PulsePrimitives.py (hunk 1 above)
Comment (blakejohnson) [inline QGL/PulsePrimitives.py]: Right, this just converts the string to a method handle.

File: QGL/config.py (hunk 2 above)
Comment (blakejohnson) [inline QGL/config.py]: Sure.

**Pull request merged**